src/providers/emoji/Emojis.cpp

```diff
@@ -12,123 +12,144 @@
+    rj::getSafe(unparsedEmoji, "non_qualified", emojiData->nonQualifiedCode);
+    rj::getSafe(unparsedEmoji, "unified", emojiData->unifiedCode);
 
-        rj::getSafe(unparsedEmoji, "has_img_apple", capabilities.apple);
-        rj::getSafe(unparsedEmoji, "has_img_google", capabilities.google);
-        rj::getSafe(unparsedEmoji, "has_img_twitter", capabilities.twitter);
-        rj::getSafe(unparsedEmoji, "has_img_facebook", capabilities.facebook);
+    rj::getSafe(unparsedEmoji, "has_img_apple", capabilities.apple);
+    rj::getSafe(unparsedEmoji, "has_img_google", capabilities.google);
+    rj::getSafe(unparsedEmoji, "has_img_twitter", capabilities.twitter);
+    rj::getSafe(unparsedEmoji, "has_img_facebook", capabilities.facebook);
 
-        if (capabilities.apple)
-        {
-            emojiData->capabilities.insert("Apple");
-        }
-        if (capabilities.google)
-        {
-            emojiData->capabilities.insert("Google");
-        }
-        if (capabilities.twitter)
-        {
-            emojiData->capabilities.insert("Twitter");
-        }
-        if (capabilities.facebook)
-        {
-            emojiData->capabilities.insert("Facebook");
-        }
+    if (capabilities.apple)
+    {
+        emojiData->capabilities.insert("Apple");
+    }
+    if (capabilities.google)
+    {
+        emojiData->capabilities.insert("Google");
+    }
+    if (capabilities.twitter)
+    {
+        emojiData->capabilities.insert("Twitter");
+    }
+    if (capabilities.facebook)
+    {
+        emojiData->capabilities.insert("Facebook");
+    }
 
-        QStringList unicodeCharacters;
-        if (!emojiData->nonQualifiedCode.isEmpty())
-        {
-            unicodeCharacters =
-                emojiData->nonQualifiedCode.toLower().split('-');
-        }
-        else
-        {
-            unicodeCharacters = emojiData->unifiedCode.toLower().split('-');
-        }
-        if (unicodeCharacters.length() < 1)
+    QStringList nonQualifiedCharacters =
+        emojiData->nonQualifiedCode.toLower().split('-');
+    QStringList unicodeCharacters = emojiData->unifiedCode.toLower().split('-');
+
+    assert(unicodeCharacters.length() >= 1);
+
+    for (const QString &unicodeCharacter : unicodeCharacters)
+    {
+        bool ok{false};
+        unicodeBytes.push_back(QString(unicodeCharacter).toUInt(&ok, 16));
+        if (!ok)
         {
+            qCWarning(chatterinoEmoji)
+                << "Failed to parse emoji" << emojiData->shortCodes;
             return;
         }
+    }
 
-        int numUnicodeBytes = 0;
+    // We can safely do a narrowing static cast since unicodeBytes will never be a large number
+    emojiData->value = QString::fromUcs4(unicodeBytes.data(),
+                                         static_cast<int>(unicodeBytes.size()));
 
-        for (const QString &unicodeCharacter : unicodeCharacters)
+    if (nonQualifiedCharacters.length() > 0)
```

Comment:
```suggestion
    if (!emojiData->nonQualifiedCode.isEmpty())
```

As noted above, there will always be one item, even in empty strings.

Reply:
Fixed in [`5198a28` (#4840)](https://github.com/Chatterino/chatterino2/pull/4840/commits/5198a287325719b7481357bd7d75b4140536e6b2)